src/metabase/models/setting.clj

```diff
@@ -1206,13 +1208,13 @@
   (default: `:no-value` for most settings; `:never` for user- and database-local settings, settings with no setter,
   and `:sensitive` settings.)"
   {:style/indent 1}
-  [setting-symbol description & {:as options}]
+  [setting-symbol description & kwargs]
```

Comment:
I'm wondering if it wouldn't be better to just make this an either-or thing, either you use kwargs like 

```clj
(defsetting my-setting
  "This is my setting"
  :x 1 :y 2)
  ```
  
  **or** you use a map of args like
  
 ```clj
(defsetting my-setting
  "This is my setting"
  {:x 1 :y 2})
  ```
  
Then you just have two different arities of this function and you don't need all the extra stuff like `expand-kwargs`

Reply:
But if you do want to keep it this way you might want to consider adding a clojure spec `fdef` for the macro, when Clojure does macroexpansion it will check the args against the spec and throw an error if they don't match

Reply:
The ergonomics I wanted relied on mixing the two, I think the readability is greatly decreased if you need to add an `assoc` onto the map, and I was not planning on supporting arbitrary forms so that wouldn't have worked.

I realized today that I had a misconception about how Clojure handles repeated keys in kwargs, and thought it was "first one wins" (i.e. most explicit), but that seems not to be the case. That means the revision you reviewed had the wrong semantics, and fixing it breaks the usage pattern I wanted as well.

Thanks for the tip on `fdef`, I haven't used clojure spec before so I'll check that out. 

With schema, malli, and clj-kondo also in the mix it seems like there are a lot of ways to generate better errors for bad macro invocations, how does one decide? I went with modifying the existing kondo linter, as catching things statically is better than at compile time - for example preventing the commit versus letting each CI test bomb out individually. 

I'd love to get general guidance on using these tools, as they're all new to me and I'm still learning their strength and weaknesses.